components/core/DataView.js

```diff
@@ -247,16 +263,95 @@ export default class DataView extends React.Component {
 
   _handleCheckScroll = Window.debounce(this._handleScroll, 200);
 
-  _handleCheckBox = (e) => {
+  _handleTableCheckBox = (e) => {
     let checked = this.state.checked;
+
+    let i = e.target.name;
+    if (this.state.isShiftDown && this.state.lastSelectedItemIndex !== i) {
+      return this._handleShiftClick({
+        currentSelectedItemIndex: i,
+        lastSelectedItemIndex: this.state.lastSelectedItemIndex,
+        checked,
+      });
+    }
+
     if (e.target.value === false) {
       delete checked[e.target.name];
-      this.setState({ checked });
+      this.setState({ checked, lastSelectedItemIndex: i });
       return;
     }
+
     this.setState({
       checked: { ...this.state.checked, [e.target.name]: true },
+      lastSelectedItemIndex: i,
     });
+
+    if (checked[i]) {
+      delete checked[i];
+    } else {
+      checked[i] = true;
+    }
+    this.setState({ checked, lastSelectedItemIndex: i });
+  };
+
+  /* (NOTE):daniel This disable text selection while pressing shift key */
+  _handleSelectStart = (e) => {
+    if (this.state.isShiftDown) {
+      e.preventDefault();
+    }
+  };
+
+  _handleKeyUp = (e) => {
```

Comment:
It looks like sometimes it isn't properly registering key up and key down events. You can see this by putting these console.logs in and looking in your console as you press and release the shift key:
![image](https://user-images.githubusercontent.com/33686587/100799620-34673c00-33da-11eb-919d-21a04938e008.png)

This is because you are checking `if(this.state.isShiftDown)`, but setting state is an asynchronous event (meaning it takes some time to finish). So if you key down then key up really quickly, it won't have enough time to set isShiftDown to true before it gets the next event to set it to false

I would recommend using a regular variable to store isShiftDown, rather than putting it in state. A regular variable won't have this issue (it sets instantly). You actually don't need to use a state variable here because isShiftDown does not affect how the page is rendered. The purpose of state variables is to ask React to "rerender the page every time this variable changes"

`this.isShiftDown = false` sets it as a regular variable as opposed to `this.setState({ isShiftDown: false })` which sets it as a part of state

Reply:
As we accumulate these hotkeys we should also start using this https://github.com/ianstormtaylor/is-hotkey

Reply:
@jimmylee is-hotkey library works with a combination of two or more keys. In this case, we are only checking if the shift key has been pressed. Hence, it will always return false in this case.